From: Doug Ledford <dledford@redhat.com>
Date: Mon, 15 Jun 2009 09:52:51 -0400
Subject: [infiniband] mlx4_en: hand remove XRC support
Message-id: ffc612e39c20333a523899ce8afda90a47c2f440.1245072810.git.dledford@redhat.com
O-Subject: [Patch RHEL5.4 16/16] [mlx4_en] hand remove XRC support from driver
Bugzilla: 506097

Signed-off-by: Doug Ledford <dledford@redhat.com>

diff --git a/drivers/net/mlx4/en_rx.c b/drivers/net/mlx4/en_rx.c
index cc27524..dc79768 100644
--- a/drivers/net/mlx4/en_rx.c
+++ b/drivers/net/mlx4/en_rx.c
@@ -506,8 +506,9 @@ int mlx4_en_activate_rx_rings(struct mlx4_en_priv *priv)
 			cpu_to_be16((i + 1) & (ring->srq.max - 1));
 		}
 	
-		err = mlx4_srq_alloc(mdev->dev, mdev->priv_pdn, ring->cqn, 0,
-				     &ring->wqres.mtt, ring->wqres.db.dma, &ring->srq);
+		err = mlx4_srq_alloc(mdev->dev, mdev->priv_pdn,
+				     &ring->wqres.mtt, ring->wqres.db.dma,
+				     &ring->srq);
 		if (err){
 			mlx4_err(mdev, "Failed to allocate srq\n");
 			ring_ind--;
@@ -521,8 +522,6 @@ int mlx4_en_activate_rx_rings(struct mlx4_en_priv *priv)
 err_srq:
 	while (ring_ind >= 0) {
 		ring = &priv->rx_ring[ring_ind];
-		mlx4_srq_invalidate(mdev->dev, &ring->srq);
-		mlx4_srq_remove(mdev->dev, &ring->srq);
 		mlx4_srq_free(mdev->dev, &ring->srq);
 		ring_ind--;
 	}
@@ -559,8 +558,6 @@ void mlx4_en_deactivate_rx_ring(struct mlx4_en_priv *priv,
 {
 	struct mlx4_en_dev *mdev = priv->mdev;
 
-	mlx4_srq_invalidate(mdev->dev, &ring->srq);
-	mlx4_srq_remove(mdev->dev, &ring->srq);
 	mlx4_srq_free(mdev->dev, &ring->srq);
 	mlx4_en_free_rx_buf(priv, ring);
 	if (ring->use_frags)
